test/powershell/engine/ETS/Adapter.Tests.ps1

```diff
@@ -80,3 +80,13 @@ Describe "Adapter Tests" -tags "CI" {
         $val | Should Be "book"
     }
 }
+
+Describe "Adapter XML Tests" -tags "CI" {
+    It "Can set XML node property to non-string object" {
+        [xml]$x  = "<root><data/></root>"
+        $rval = @{c=1}
+        # rval will be implicitly converted to `string` type
+        { $x.root.data = $rval } | Should Not Throw
+        $x.root.data | Should Be $rval.ToString()
+    }

```

Comment:
You should also test that we properly invoke `ToString` implemented as a `CodeMethod` and `ScriptMethod`. `System.Type` has a `CodeMethod` implementation of `ToString`, you can see the difference:
```
#57 PS> [System.Int32].ToString()
System.Int32
#58 PS> [string][System.Int32]
int
```